scripts/Crawlers/BaseCrawler.py

```diff
@@ -0,0 +1,412 @@
+    ==================
+
+    Any crawler created from this interface will have to crawl
+    datasets from a specific remote platforms. This base class
+    implements the functions common to all crawled backends, in particular:
+    (1) verify that correct fork of conp-dataset is used,
+    (2) create and switch to an new branch for each new dataset,
+    (3) ignore README and DATS files for the annex,
+    (4) create new datalad datasets,
+    (5) publish to GitHub repository,
+    (6) create pull requests.
+
+    Method run(), implemented in the base class, is the entry point to any crawler.
+    It does the following things:
+    (1) It calls abstract method get_all_dataset_description(), that must be implemented
+    by the crawler in the child class. get_all_dataset_description()
+    retrieves from the remote platform
+    all the necessary information about each dataset that is supposed to be added or
+    updated to conp-dataset.
+    (2) It iterates through each dataset description, and switch to a dedicated git branch
+    for each dataset.
+       (2.a) If the dataset is new, the base class will create a new branch,
+             an empty datalad repository, unannex DATS.json and README.md and create an 
+             empty GitHub repository. It will then call abstract method add_new_dataset()
+             which will add/download all dataset files under given directory.
+             The crawler will then add a custom DATS.json and README.md if those weren't added.
+             Creating the README.md requires get_readme_content() to be implemented, which will
+             return the content of the README.md in markdown format. The crawler will then save and
+             publish all changes to the newly create repository. It will also handle adding a new submodule
+             to .gitmodules and creating a pull request to CONP-PCNO/conp-dataset.
+       (2.b) If the dataset already exists, verified by the existence of its corresponding branch,
+             the base class will call abstract method update_if_necessary() which will verify
+             if the dataset requires updating and update if so. If the dataset got updated, This method
+             will return True which will trigger saving, publishing new content to the dataset's respective
+             repository, creating a new DATS.json if it doesn't exist and creating a pull 
+             request to CONP-PCNO/conp-dataset.
+
+    ==================
+    How to implement a new crawler
+    ==================
+
+        (1) Create a class deriving from BaseCrawler
+        (2) Implement the four abstract methods:
+            * get_all_dataset_description,
+            * add_new_dataset
+            * update_if_necessary
+            * get_readme_content.
+            See docstrings of each method for specifications.
+        (3) In crawl.py, locate the comment where it says to instantiate new crawlers,
+            instantiate this new Crawler and call run() on it
```

Comment:
Your text was good, I reformulated slightly. To explain complex things in code comments, I recommend to be as structured as possible:
* Add titles
* Make bullet/numbered lists
* Make short sentences